runtime/vm/OutOfLineINL_java_lang_invoke_NativeMethodHandle.cpp

```diff
@@ -135,6 +134,13 @@ OutOfLineINL_java_lang_invoke_NativeMethodHandle_initJ9NativeCalloutDataRef(J9VM
 	nativeCalloutData->arguments = args;
 	nativeCalloutData->cif = cif;
 
+	/* Hash table in systemClassLoader is used to cleanup J9NativeCalloutData when the classLoader is unloaded */
+	if (NULL == hashTableAdd(vm->systemClassLoader->nativeCalloutDataHashTable, nativeCalloutData)) {
```

Comment:
get the classloader from the `methodHandle` class, instead of directly looking up the systemClassloader. 

99% of the time it will be the systemClassLoader, however it is possible to construct a case where this is not true